dev/forbidden_from_release_tests/bin/main.dart

```diff
@@ -0,0 +1,272 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:path/path.dart' as path;
+import 'package:process/process.dart';
+
+const ProcessManager processManager = LocalProcessManager();
+const FileSystem fs = LocalFileSystem();
+
+Future<void> main(List<String> args) async {
+  final Options options = Options.fromArgs(args);
+
+  final Directory tempDirectory = fs.systemTempDirectory.createTempSync('forbidden_imports');
+  print('Using $tempDirectory for temporary files.');
+
+  await pubGet(options.flutter, options.target.parent);
+  final File dill = await compileKernel(
+    tempDirectory: tempDirectory,
+    dart: options.dart,
+    target: options.target,
+    feSnapshot: options.feSnapshot,
+    sdkProduct: options.sdkProduct,
+    packageConfig: options.packageConfig,
+    packageName: options.packageName,
+  );
+
+  final File v8SnapshotInfo = await compileAOTSnapshot(
+    tempDirectory: tempDirectory,
+    genSnapshot: options.genSnapshot,
+    dill: dill,
+  );
+
+  final String json = v8SnapshotInfo.readAsStringSync();
+  final List<String> foundForbiddenTypes = <String>[];
+  for (final String forbiddenType in options.forbiddenTypes) {
+    if (json.contains(forbiddenType)) {
+      foundForbiddenTypes.add(forbiddenType);
+    }
+  }
+  if (foundForbiddenTypes.isNotEmpty) {
+    print('The output contained the following forbidden types:');
+    print(foundForbiddenTypes);
+    exit(-1);
+  }
+
+  tempDirectory.deleteSync(recursive: true);
+}
+
+Future<void> pubGet(File flutter, Directory target) async {
+  final List<String> command = <String>[flutter.path, 'pub', 'get'];
+  print('Running command: ${command.join(' ')} in ${target.path}');
+  final ProcessResult pubGetResult = await processManager.run(
+    command,
+    workingDirectory: target.path,
+  );
+  if (pubGetResult.exitCode != 0) {
+    print(pubGetResult.stdout);
+    print('Error running pub get.');
+    print(pubGetResult.stderr);
+    exit(pubGetResult.exitCode);
+  }
+}
+
+Future<File> compileKernel({
+  required Directory tempDirectory,
+  required File dart,
+  required File target,
+  required File feSnapshot,
+  required Directory sdkProduct,
+  required File packageConfig,
+  required String packageName,
+}) async {
+  final File dill = tempDirectory.childFile('forbidden_release_imports.dill');
```

Comment:
This is going to be awkward to keep in sync with changes in the tool. Perhaps we could make build bundle support outputting release kernel? Otherwise we need something automated to keep this in sync.

I'm worried about us updating the flags and then forgetting to update this as well, leading to mistaken test results.

Reply:
I'm less concerned about building the kernel as being about to get a deterministic path for the snapshot summary file.

Reply:
To be perfectly clear, I'm not promising to remember this test or consider updates to the flags we pass to the frontend_server to be breaking in anyway. If you're OK remembering to keep this up to date on your own, then by all means :)

Reply:
This has been refactored to just use `flutter build apk`.